From 48aaa2fc53f1be91c799f984dc1ee5f384c9640f Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Wed, 31 May 2023 05:48:35 +0200
Subject: [PATCH 1/6] cursor: Always follow cursor position from mir

It makes no sense to not, if we dont we can quickly get out of sync with
mir.

This was done for multiscreen, but this has to be fixed another way then
tracking pointer ourself. The plan is to follow mir's screen position
anyway so we don't need this. (the current multiscreen impl is horrable)
---
 plugins/Cursor/InputDispatcherFilter.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plugins/Cursor/InputDispatcherFilter.cpp b/plugins/Cursor/InputDispatcherFilter.cpp
index 4ef0199d9..5dcc1a421 100644
--- a/plugins/Cursor/InputDispatcherFilter.cpp
+++ b/plugins/Cursor/InputDispatcherFilter.cpp
@@ -72,7 +72,8 @@ bool InputDispatcherFilter::eventFilter(QObject *o, QEvent *e)
             QPointF movement = me->localPos();
 
             // Adjust the position
-            QPointF oldPos = pointer->window()->geometry().topLeft() + pointer->position();
+            // Get the pos from mir
+            QPointF oldPos = me->screenPos();
             QPointF newPos = adjustedPositionForMovement(oldPos, movement);
 
             QScreen* currentScreen = screenAt(newPos);
-- 
2.46.0

